src/xmipp/libraries/reconstruction/angular_assignment_mag.h

```diff
@@ -0,0 +1,212 @@
+    void readParams();
+
+    void show();
+
+    void startProcessing();
+
+    void preProcess();
+
+    void processImage(const FileName &fnImg, const FileName &fnImgOut, const MDRow &rowIn, MDRow &rowOut);
+
+    void postProcess();
+
+    void applyCircularMask(const MultidimArray<double> &in, MultidimArray<double> &out);
+    void applyFourierImage(MultidimArray<double> &data, MultidimArray<std::complex<double> > &FourierData);
+    void applyFourierImage(MultidimArray<double> &data, MultidimArray<std::complex<double> > &FourierData, const size_t &ang);
+    void applyFourierImage2(MultidimArray<double> &data, MultidimArray<std::complex<double> > &FourierData);
+    void applyFourierImage2(MultidimArray<double> &data, MultidimArray<std::complex<double> > &FourierData, const size_t &ang);
+    void applyFourierImage3(MultidimArray<double> &data, MultidimArray<std::complex<double> > &FourierData, const size_t &ang);
+    void applyRotation(const MultidimArray<double> &MDaRef, double &rot, MultidimArray<double> &MDaRefRot);
+    void applyRotation(MultidimArray<double> &MDaRef, double &rot, MultidimArray<double> &MDaRefRot);
+    void applyShift(MultidimArray<double> &input, double &tx, double &ty, MultidimArray<double> &output);
+    void applyShift(const MultidimArray<double> &input, double &tx, double &ty, MultidimArray<double> &output);
+    void applyRotationAndShift(const MultidimArray<double> &MDaRef, double &rot, double &tx, double &ty, MultidimArray<double> &MDaRefRot);
+    void applyShiftAndRotation(const MultidimArray<double> &MDaRef, double &rot, double &tx, double &ty, MultidimArray<double> &MDaRefRot);
+
+    void bestCand(const MultidimArray<double> &MDaIn, const MultidimArray<std::complex<double> > &MDaInF, const MultidimArray<double> &MDaRef, std::vector<double> &cand, double &bestCandRot, double &shift_x, double &shift_y, double &bestCoeff);
+
+    void completeFourierShift(MultidimArray<double> &in, MultidimArray<double> &out);
+    void ccMatrix(const MultidimArray<std::complex<double> > &F1, const MultidimArray<std::complex<double> > &F2, MultidimArray<double> &result);
+    void computingNeighborGraph();
+    void computeLaplacianMatrix(Matrix2D<double> &L, const std::vector< std::vector<int> > &allNeighborsjp, const std::vector< std::vector<double> > &allWeightsjp);
+    void computeCircular();
+    void circularWindow(MultidimArray<double> &in);
+
+    void getComplexMagnitude(MultidimArray<std::complex<double> > &FourierData, MultidimArray<double> &FourierMag);
+    void getShift(MultidimArray<double> &ccVector, double &shift, const size_t &size);
+    void graphFourierFilter(Matrix1D<double> &ccVecIn, Matrix1D<double> &ccVecOut);
+
+    MultidimArray<double> imToPolar(MultidimArray<double> &cartIm, size_t &start, size_t &end);
+    double interpolate(MultidimArray<double> &cartIm, double &x_coord, double &y_coord);
+
+    void meanByColumn(MultidimArray<double> &in, MultidimArray<double> &out);
```

Comment:
where you can, pass objects by `const` reference, e.g. `void meanByColumn(const MultidimArray<double> &in, MultidimArray<double> &out);`

Reply:
Done